vpr/src/base/partition.h

```diff
@@ -0,0 +1,47 @@
+#ifndef PARTITION_H
+#define PARTITION_H
+
+#include "vtr_strong_id.h"
+#include "region.h"
+#include "partition_regions.h"
+
+/**
+ * @file
+ * @brief This file defines the data for a partition: a grouping of atoms that are constrained to a portion of an FPGA. A partition defines
+ * the atoms that are assigned to it, and the locations in which they can be placed. The locations are a union of rectangles in x, y,
+ * sub_tile space. Common cases include a single rectangle or a single x, y, sub_tile (specific location). More complex partitions
+ * with L, T or other shapes can be created with a union of multiple rectangles.
+ */
+
+//Type tag for PartitionId
+struct partition_id_tag;
+
+//A unique identifier for a partition
+typedef vtr::StrongId<partition_id_tag> PartitionId;
+
+class Partition {
+  public:
```

Comment:
have a member function to see whether a block is locked to a specific x, y, subtile location - comment can be called by placer to see if it can be immediately locked down

Reply:
Done.